website/docs/file_sizing.md

```diff
@@ -3,51 +3,76 @@ title: "File Sizing"
-clustering comes to the rescue. Clustering can be scheduled through the ingestion job and an asynchronus job can stitch 
-small files together in the background to generate larger files. NOTE that during this, ingestion can continue to run concurrently.
-
-*Please note that Hudi always creates immutable files on disk. To be able to do auto-sizing or clustering, Hudi will 
-always create a newer version of the smaller file, resulting in 2 versions of the same file. 
-The [cleaner service](/docs/next/hoodie_cleaner) will later kick in and delete the older version small file and keep the latest one.*
\ No newline at end of file
+A fundamental problem when writing data to a source is having a lot of small files. This is also known as a small file problem. If you don’t size the files appropriately, you can slow down the query performance and work with stale analytics. Some of the issues you may encounter with small files include the following:
+- **Reads slow down**: You’ll have to scan through many small files to retrieve data for a query. It’s a very inefficient way of accessing and utilizing the data.
+
+- **Processes slow down**: You can slow down your i.e., Spark or Hive jobs; the more files you have, the more tasks you create for reading each file.
+
+- **Storage increases**: When working with a lot of data, you can be inefficient in using your storage. For example, many small files can have a lower compression ratio, leading to more data on storage. If you’re indexing the data, that also takes up more storage space inside the Parquet files. If you’re working with a small amount of data, you might not see a significant impact with storage. However, when dealing with petabyte and exabyte data, you’ll need to be efficient in managing storage resources. 
+
+All these challenges inevitably lead to stale analytics and scalability challenges:
+- Query performance slows down.
+- Jobs could be running faster.
+- You’re utilizing way more resources. 
+
+A critical design decision in the Hudi architecture is to avoid small file creation. Hudi is uniquely designed to write appropriately sized files automatically. This document will show you how Apache Hudi overcomes the dreaded small files problem. There are two ways to manage small files in Hudi: 
+
+- Auto-size during ingestion
+- Clustering
+
+Below, we will describe the advantages and trade-offs of each.
+
+## Auto-sizing during ingestion​
+
+You can manage file sizes through Hudi’s auto-sizing capability during ingestion. The default targeted file size for Parquet base files is 120MB, which can be configured by hoodie.parquet.max.file.size. Auto-sizing may add some data latency, but it ensures that the read queries are always efficient as soon as a write transaction is committed. It’s important to note that if you don’t manage file sizing as you write and,  instead, try to run clustering to fix your file sizing periodically, your queries might be slow until the point when the clustering finishes.
+
```

Comment:
`,  instead` -> `, instead`

Reply:
In general, the clustering does not support data UPSERTs yet, not sure if it is appropiate to address enabling the clustering here, because clustering only solves the small file issues in APPEND ONLY senarios.

Reply:
I think it's ok, to add that at the end and say refer to the clustering docs for more details?

Reply:
Yeah, should be fine